plugins/tech-insights-node/src/facts.ts

```diff
@@ -174,6 +174,16 @@ export interface FactRetriever {
    */
   version: string;
 
+  /**
+   * A short display title for the fact retriever to be used in the interface
+   */
+  title: string;
+
+  /**
+   * A short display description for the fact retriever to be used in the interface.
+   */
+  description: string;
+
```

Comment:
Perhaps even makes sense to make these optional? Feels like they have reasonable fallbacks. That way it's not even a breaking change

Reply:
Yea I considered it but not sure what the fallbacks would be and it would break the UI a bit if they were not present. Also at this stage I feel its better to make things required as its not going to be so painful and it reduces code complexity of handling undefined fields in future.

Reply:
Saying that, I'll make them optional as I guess there's no current use case to expose fact retrievers in the frontend